sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -59,6 +59,25 @@ def predict(self, X):
         return np.ones(X.shape[0])
 
 
+class NoSampleWeightPandasSeriesType(BaseEstimator):
+    def fit(self, X, y, sample_weight=None):
+        # Convert data
+        X, y = check_X_y(X, y,
+                         accept_sparse=("csr", "csc"),
+                         multi_output=True,
+                         y_numeric=True)
+        # Loosely based on _solve_cholesky_kernel (called in KernelRidge.fit)
```

Comment:
This is far too confusing. Just raise an exception directly if sample_weight is a pandas Series.